Fix printing summaries of multiindex coords

coroa opened the pull request:
Since merging #2293 summarize_variable displays the first and last few entries,
so we have to pass all along.

MWE:
```python
import xarray as xr
import numpy as np
x = np.arange(0, 90, 0.5)
y = np.arange(0, 90, 0.5)
da = xr.DataArray(np.random.random((len(x), len(y))), [('x', x), ('y', y)])
da.stack(z=('x', 'y'))
```
gives on master:
```
<xarray.DataArray (z: 32400)>
array([0.497724, 0.793955, 0.140193, ..., 0.05734 , 0.884105, 0.479093])
Coordinates:
  * z        (z) MultiIndex
  - x        (z) float64 0.0 0.0 0.0 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0 0.0 0.0
  - y        (z) float64 0.0 0.5 1.0 1.5 2.0 2.5 ... 12.5 13.0 13.5 14.0 14.5
```
and on this branch
```
<xarray.DataArray (z: 32400)>
array([0.497724, 0.793955, 0.140193, ..., 0.05734 , 0.884105, 0.479093])
Coordinates:
  * z     (z) MultiIndex
  - x     (z) float64 0.0 0.0 0.0 0.0 0.0 0.0 ... 89.5 89.5 89.5 89.5 89.5 89.5
  - y     (z) float64 0.0 0.5 1.0 1.5 2.0 2.5 ... 87.0 87.5 88.0 88.5 89.0 89.5
```

 - [ ] Closes #xxxx
 - [x] Tests added
 - [ ] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

shoyer commented:
Thanks @coroa. Would you mind adding a test to verify your fix? It could probably go alongside `test_repr_multiindex`:
https://github.com/pydata/xarray/blob/23fc0d6f8a5affefb84e5f20dc0bef7ace4542e2/xarray/tests/test_dataarray.py#L57-L65

shoyer commented:
The test failures on pandas-dev are unrelated. If you merge in master, I believe they should be fixed.

pep8speaks commented:
Hello @coroa! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`xarray/tests/test_dataarray.py`](https://github.com/pydata/xarray/blob/e3176d3115cd2a45758356d6197194f5a3528ff7/xarray/tests/test_dataarray.py):

> [Line 41:80](https://github.com/pydata/xarray/blob/e3176d3115cd2a45758356d6197194f5a3528ff7/xarray/tests/test_dataarray.py#L41): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 42:80](https://github.com/pydata/xarray/blob/e3176d3115cd2a45758356d6197194f5a3528ff7/xarray/tests/test_dataarray.py#L42): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 76:80](https://github.com/pydata/xarray/blob/e3176d3115cd2a45758356d6197194f5a3528ff7/xarray/tests/test_dataarray.py#L76): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 80:80](https://github.com/pydata/xarray/blob/e3176d3115cd2a45758356d6197194f5a3528ff7/xarray/tests/test_dataarray.py#L80): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 81:80](https://github.com/pydata/xarray/blob/e3176d3115cd2a45758356d6197194f5a3528ff7/xarray/tests/test_dataarray.py#L81): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)

coroa commented:
Ok, rebased on master, added tests. Was not sooo easy to make linter and pytest happy :), but we're there. Hope skipping `E501 line too long (... > 79 characters)` for the multiline test string is allowed.

https://github.com/pydata/xarray/blob/e4716a4c2b9df681eb9e43dd31a3c9ec6e104e44/xarray/tests/test_dataarray.py#L74-L81

shoyer commented on xarray/tests/test_dataarray.py:
```diff
@@ -37,6 +37,12 @@ def setup(self):
                                                  names=('level_1', 'level_2'))
         self.mda = DataArray([0, 1, 2, 3], coords={'x': self.mindex}, dims='x')
 
+        self.mindex_long = pd.MultiIndex.from_product(
```
Would you mind only defining these variables in the `test_repr_multiindex_long` method? 

We try to only define variables in `setup()` that are used in multiple methods (and even then, it's often better to just repeat setup code)

coroa replied:
Sure enough, was my initial thought as well, but since `self.{mindex,mda}` were only used once and defined in the fixture, i had arranged it like that.

coroa replied:
Done

dcherian commented:
@coroa you should give yourself credit in `whats-new.rst`

shoyer commented:
thanks @coroa! this will be in the next xarray release
